src/common/compress/lz4/decompress.c

```diff
@@ -0,0 +1,192 @@
+    bool frameDone;                                                 // Has the current frame completed?
+    bool done;                                                      // Is decompression done?
+} Lz4Decompress;
+
+/***********************************************************************************************************************************
+Render as string for logging
+***********************************************************************************************************************************/
+static String *
+lz4DecompressToLog(const Lz4Decompress *this)
+{
+    return strNewFmt(
+        "{inputSame: %s, inputOffset: %zu, frameDone %s, done: %s}", cvtBoolToConstZ(this->inputSame), this->inputOffset,
+        cvtBoolToConstZ(this->frameDone), cvtBoolToConstZ(this->done));
+}
+
+#define FUNCTION_LOG_LZ4_DECOMPRESS_TYPE                                                                                           \
+    Lz4Decompress *
+#define FUNCTION_LOG_LZ4_DECOMPRESS_FORMAT(value, buffer, bufferSize)                                                              \
+    FUNCTION_LOG_STRING_OBJECT_FORMAT(value, lz4DecompressToLog, buffer, bufferSize)
+
+/***********************************************************************************************************************************
+Free decompression context
+***********************************************************************************************************************************/
+OBJECT_DEFINE_FREE_RESOURCE_BEGIN(LZ4_DECOMPRESS, LOG, logLevelTrace)
+{
+    LZ4F_freeDecompressionContext(this->context);
+}
+OBJECT_DEFINE_FREE_RESOURCE_END(LOG);
+
+/***********************************************************************************************************************************
+Decompress data
+***********************************************************************************************************************************/
+static void
+lz4DecompressProcess(THIS_VOID, const Buffer *compressed, Buffer *decompressed)
+{
+    THIS(Lz4Decompress);
+
+    FUNCTION_LOG_BEGIN(logLevelTrace);
+        FUNCTION_LOG_PARAM(LZ4_DECOMPRESS, this);
+        FUNCTION_LOG_PARAM(BUFFER, compressed);
+        FUNCTION_LOG_PARAM(BUFFER, decompressed);
+    FUNCTION_LOG_END();
+
+    ASSERT(this != NULL);
+    ASSERT(this->context != NULL);
+    ASSERT(decompressed != NULL);
+
+    // When there is no more input then decompression is done
+    if (compressed == NULL)
+    {
+        // If the current frame being decompressed was not completed then error
+        if (!this->frameDone)
+            THROW(FormatError, "unexpected eof in compressed data");
+
+        this->done = true;
+    }
+    else
+    {
+        // Decompress as much data as possible
+        size_t srcSize = bufUsed(compressed) - this->inputOffset;
+        size_t dstSize = bufRemains(decompressed);
+
+        this->frameDone = !lz4Error(
```

Comment:
Could we make this == 0 or 0 ==?

Reply:
Went with == 0.  0 == this looks too weird to me.